federation/pkg/federation-controller/ingress/ingress_controller.go

```diff
@@ -624,10 +624,7 @@ func (ic *IngressController) reconcileIngress(ingress types.NamespacedName) {
 			ic.deliverIngress(ingress, 0, true)
 			return
 		}
-		desiredIngress := &extensions_v1beta1.Ingress{
-			ObjectMeta: baseIngress.ObjectMeta,
-			Spec:       baseIngress.Spec,
-		}
+		desiredIngress := &extensions_v1beta1.Ingress{}
 		objMeta, err := conversion.NewCloner().DeepCopy(baseIngress.ObjectMeta)
```

Comment:
No, we don't always overwrite it.  But I agree, the code is confusing, and in need ot refactoring.  Will do that holistically in a followup PR.